docs/typescript/reference/configuration.md

```diff
@@ -1,186 +1,126 @@
+- **logLevel**: Configure the [DBOS logger](../tutorials/logging.md) severity. Defaults to `info`.
+- **otlpTracesEndpoints**: DBOS operations [automatically generate OpenTelemetry Traces](../tutorials/logging.md). Use this field to declare a list of OTLP-compatible receivers.
+- **runAdminServer**: Whether to run an [HTTP admin server](../../production/self-hosting/admin-api.md) for workflow management operations. Defaults to True.
+- **adminPort**: The port on which the admin server runs. Defaults to 3001.
 
-This section is used to specify DBOS HTTP parameters. Only used if you are using the DBOS HTTP decorators.
 
-- **cors_middleware** (optional): Whether to allow cross-origin requests (via CORS middleware).  Default is `true`.
-- **credentials** (optional): If `cors_middleware` is enabled, whether to allow cross-origin requests that include credentials.  Default is `true`.
-- **allowed_origins** (optional): If `cors_middleware` is enabled, a list of origins/domains that are permitted to make cross-origin requests.  Default is to allow any origin.
+## DBOS Configuration File
 
-**Example**:
+Many tools in the DBOS ecosystem are configured by a `dbos-config.yaml` file.
+Tools that use `dbos-config.yaml` include the [DBOS CLI](./tools/cli.md), [DBOS debugger](../tutorials/debugging.md), and [DBOS Cloud](../../production/dbos-cloud/deploying-to-cloud.md).
+Additionally, the DBOS library will fall back to `dbos-config.yaml` if no configuration object is provided.
 
-```yaml
-http:
-  cors_middleware: true
-  credentials: true
-  allowed_origins:
-    - 'https://partner.com'
-    - 'https://app.internal.com'
-```
----
-
-### Application
+Here is an example configuration file with default parameters:
 
-Applications can optionally use the application configuration to define custom properties as key-value pairs.
-These properties can be retrieved from any [context](./transactapi/oldapi/contexts) via the [`getConfig`](./transactapi/oldapi/contexts#ctxtgetconfig) method.
-
-**Example**:
-```yaml
-application:
-  PAYMENTS_SERVICE: 'https://stripe.com/payment'
+```shell
+name: my-app
+language: node
+database_url: ${DBOS_DATABASE_URL}
+runtimeConfig:
+  start:
+    - node dist/main.js
 ```
 
-### Environment Variables
-
-Applications can optionally use the `env` configuration to define environment variables.
-These are set in your application before its code is initialized and can be retrieved from `process.env` like any other environment variables.
-For example, the `WEB_PORTAL` variable set below could be retrieved from an application as `process.env.WEB_PORTAL`.
-Environment variables configured here are automatically exported to and set in DBOS Cloud.
+### Configuration File Fields
 
-**Example**:
-```yaml
-env:
-  WEB_PORTAL: 'https://example.com'
-```
+::::info
+You can use environment variables for configuration values through the syntax `field: ${VALUE}`.
+::::
```

Comment:
we shouldn't advertise this anymore

Reply:
It's literally unusable without this (for example, $DBOS_DATABASE_URL in all our demo apps)